### ISSUE TITLE
Punctuation is no longer a candidate for link hints

### ISSUE THREAD

**Issue by jlreeder** [philc/vimium#2150]
Thanks for your work. I'm a huge fan of Vimium.

With `link-hint mode` enabled, in an earlier version of Vimium, if a link consistent of punctuation, eg `"`, I could navigate to it by typing <kbd>f</kbd> + <kbd>"</kbd>. In Vimium 1.55, typing <kbd>"</kbd> after <kbd>f</kbd> has no effect.

This functionality may seem unimportant, as very few actual links consist of only punctuation. But its use is clearer if you consider that buttons are also links. In my case, I work with a webtool that has a whole slew of punctuation buttons: `"`, `'`,  `;`. I hope you'll consider reinstating the feature.

Versions:
- Google Chrome: 50.0.2661.102 (Official Build) (64-bit)
- Vimium: 1.55

**Comment by smblott-github** [comment]
Thanks, @jlreeder.  Your point is valid, however there's a reason for the change.

With filtered link hints, Vimium scores the matches and tries to prioritise the best match.  Matches get higher scores at the start of words, for whole words, and highest of all at the start of the link text.

If the user starts typing the start of the link text, then the intended link is often selected after just a few keystrokes, and `Enter` selects that link.

To see this in effect, scroll to the top of this page at enter `f` `n`.  Observe that "New Issue" is selected.  So, `f` `n` `Enter` activates that link.  And it's quite predictable.

If the link text is `"new issue"` (including quotes), then typing `n` will give the link a very high score, whereas typing `s` will give a low score.  If we included the `"` in the matching text, then which character is the start of the word?  `n` or `"`?

To summarise, the change was made to ensure that typing `n` (in the example above) would give a very high score.  (It wasn't previously.)

Now, I'm sure it would be possible to get the behaviour you want and the behaviour I want together.  Let me think about that.

**Comment by jlreeder** [comment]
Thanks for that detailed explanation. It sounds like you've understood my point.

I do agree that it's important for typing <kbd>f</kbd> <kbd>n</kbd> to give a high score for the link `"new issue"`, and I even think that is a higher priority than allowing <kbd>f</kbd>  <kbd>"</kbd> to give a high score, since my guess is most people ignore quotes and other punctuation when typing link hints.

As far I can tell, now <kbd>f</kbd>  <kbd>"</kbd> is having no effect whatsoever. Perhaps the compromise is to allow punctuation matches to add to the match score, but at some fraction of the amount that letters do. I'll defer to your judgment and prioritization there.
